### DISCUSSION TITLE
customerCreate should return an access token

### DISCUSSION THREAD

**Discussion by nboliver** [Shopify/storefront-api-feedback#140] [Ideas & Requests]
The `CustomerCreatePayload` should return an access token when there are no errors creating the customer. This would simplify the flow for newly registered customers and avoid having to make a second API call to get a token, since the optimal UX would be to log the customer in after they have registered successfully.

Found this [comment](https://github.com/Shopify/hydrogen/blob/v1.x-2022-07/templates/demo-store/src/components/account/AccountCreateForm.client.tsx#L56) in the Hydrogen demo repo; the code author seems to agree.

**Comment by vixdug** [comment]
Hey @nboliver - Once you create the account, you will be activating it as well. The activation mutation returns the access token: https://shopify.dev/api/storefront/2022-10/mutations/customerActivate#field-customeractivatepayload-customeraccesstoken

**Reply by nboliver** [reply]
@vixdug That makes sense for existing customers that already have an account, since they need to go through the activation flow / confirm their email (`customerCreate` will return the `CUSTOMER_DISABLED` `customerUserError` in this case). 
For new customers that are creating an account, there is no activation flow - they can access the account immediately. In this case, we are passing the form data (email / password) that they used when filling out the `customerCreate` form as input variables back to `customerAccessTokenCreate` in order to get a token, but it requires an extra API call that shouldn't be needed.

**Reply by vixdug** [reply]
Good point, that is a good distinction - I appreciate the feedback.
